saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationRequestFilter.java

```diff
@@ -91,39 +91,50 @@ private void sendRedirect(HttpServletRequest request, HttpServletResponse respon
 	}
 
 	private String createSamlRequestRedirectUrl(HttpServletRequest request, RelyingPartyRegistration relyingParty) {
-		Saml2AuthenticationRequest authNRequest = createAuthenticationRequest(relyingParty, request);
-		String xml = this.authenticationRequestFactory.createAuthenticationRequest(authNRequest);
-		String encoded = encode(deflate(xml));
-		String relayState = request.getParameter("RelayState");
-		UriComponentsBuilder uriBuilder = UriComponentsBuilder
-				.fromUriString(relyingParty.getIdpWebSsoUrl())
-				.queryParam("SAMLRequest", UriUtils.encode(encoded, StandardCharsets.ISO_8859_1));
-
-		if (StringUtils.hasText(relayState)) {
-			uriBuilder.queryParam("RelayState", UriUtils.encode(relayState, StandardCharsets.ISO_8859_1));
+		Saml2AuthenticationRequest authnRequest = createAuthenticationRequest(relyingParty, request, true);
+		MultiValueMap<String, String> qp =
+				this.authenticationRequestFactory
+						.createAuthenticationRequestParameters(authnRequest)
+						.getQueryParameters();
+
+		UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUriString(relyingParty.getIdpWebSsoUrl());
```

Comment:
It might help prove out the contract if a line was drawn here between the relying party, the request, and the final Saml2AuthenticationRequest instance. For example, I'd probably do `authnRequest.getDestination()` instead of `relyingParty.getIdpWebSsoUrl()`.

Reply:
Addressed in: 83e9929fb3a2b9225f64e1fe31f6590c47bc2320